src/persistence/tree/MaterializedPathSubjectExecutor.ts

```diff
@@ -53,4 +53,46 @@ export class MaterializedPathSubjectExecutor {
             .execute();
     }
 
+    /**
+     * Executes operations when subject is being updated.
+     */
+    async update(subject: Subject): Promise<void> {
+        let parent = subject.metadata.treeParentRelation!.getEntityValue(subject.entity!); // if entity was attached via parent
+
+        const parentId = subject.metadata.getEntityIdMap(parent);
+
+        let parentPath: string = "";
+        if (parentId) {
+            parentPath = await this.queryRunner.manager
+                .createQueryBuilder()
+                .select(subject.metadata.targetName + "." + subject.metadata.materializedPathColumn!.propertyPath, "path")
+                .from(subject.metadata.target, subject.metadata.targetName)
+                .whereInIds(parentId)
+                .getRawOne()
+                .then(result => result ? result["path"] : undefined);
+        }
+
+        const updatedEntityId = subject.metadata.treeParentRelation!.joinColumns.map(joinColumn =>
+            joinColumn.referencedColumn!.getEntityValue(subject.entity!)).join("_");
+
+        const materializedPathColumn = subject.metadata.materializedPathColumn!.propertyPath;
+        const oldPath = await this.queryRunner.manager
+            .createQueryBuilder()
+            .select(materializedPathColumn, "path")
+            .from(subject.metadata.target, subject.metadata.targetName)
+            .where(subject.identifier!)
+            .getRawOne()
+            .then(result => result ? result["path"] : undefined);
+
+        const newPath = parentPath + updatedEntityId + ".";
+        await this.queryRunner.manager
+            .createQueryBuilder()
+            .update(subject.metadata.target)
+            .set({ [materializedPathColumn]: () => `REPLACE(${materializedPathColumn}, :oldPath, :newPath)` } as any)
```

Comment:
Is it possible that this replaces the wrong location? Because the substring for oldPath could occur anywhere in the string.  Perhaps substr to chop off the prefix that we know is there because it matched in the where clause and concat the new prefix?